apps/s_client.c

```diff
@@ -1742,7 +1762,11 @@ int s_client_main(int argc, char **argv)
     if (fallback_scsv)
         SSL_set_mode(con, SSL_MODE_SEND_FALLBACK_SCSV);
 
-    if (servername != NULL) {
+    if (!noservername && (servername != NULL || dane_tlsa_domain == NULL)) {
+        if (servername == NULL)
+            servername = host;
```

Comment:
can combine these three lines into a single ```?:``` assignment. or  embed the second null test inside the if.